### PR TITLE
Add independent private plan farming pool account

### PR DESCRIPTION
**Pull request by dongsam**
## Description

closes: #58 

## Bullet Points

- [x] fix proto for plan creation msgs ( from FarmingPoolAddr to Creator )
- [x] add functions for generation `PrivatePlanFarmingPoolAcc`
- [x] fix bugs on PublicPlanProposal handlers ( on creation, termination address )
- [x] fix logics for msg handlers
- [x] fix logics for proposal handlers
- [x] cli tx
- [x] add testcodes
- [x] update spec docs
- [x] update relevant docs

## References

- 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by dongsam** [comment]
Can you review the logic before working on the document update? @kogisin

File: x/farming/keeper/plan.go [hunk 1]
```diff
@@ -170,37 +170,35 @@ func (k Keeper) UnmarshalPlan(bz []byte) (plan types.PlanI, err error) {
...
-	terminationAddrAcc := farmingPoolAddrAcc
-
-	params := k.GetParams(ctx)
-
-	balances := k.bankKeeper.GetAllBalances(ctx, farmingPoolAddrAcc)
-	diffs, hasNeg := balances.SafeSub(params.PrivatePlanCreationFee)
-	if hasNeg {
-		return nil, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFunds, "lack of %s coins to pay private plan createion fee", diffs.String())
-	}
+	// TODO: if public proposal, It should be choose custom terminationAddr
```
Comment (kogisin) [inline x/farming/keeper/plan.go:175]: Can you elaborate more about this comment? If the plan type is public proposal, we should choose custom `terminationAcc`? 

You modify `CreateFixedAmountPlan` to receive `terminationAcc` as one parameter and use `terminationAcc.String()` when defining `basePlan` variable.

File: x/farming/keeper/plan.go [hunk 2]
```diff
@@ -170,37 +170,35 @@ func (k Keeper) UnmarshalPlan(bz []byte) (plan types.PlanI, err error) {
...
+
+	// TODO: check if private, not equal msg.creator, terminationAcc
```
Comment (kogisin) [inline x/farming/keeper/plan.go:177]: Is this comment for whether or not we should add validation logic for `msg.creator` is equal to `terminationAcc`?

File: x/farming/keeper/proposal_handler.go [hunk 3]
```diff
@@ -42,6 +42,10 @@ func (k Keeper) AddPublicPlanProposal(ctx sdk.Context, proposals []*types.AddReq
 		if err != nil {
 			return err
 		}
+		terminationAcc, err := sdk.AccAddressFromBech32(p.TerminationAddress)
```
Comment (kogisin) [inline x/farming/keeper/proposal_handler.go:45]: Just a nitpick around using the same way as the right above logic. I suggest to use `p.GetTerminationAddress()` or change `p.GetFarmingPoolAddress()` to `p.FarmingPoolAddress`.

File: x/farming/keeper/proposal_handler_test.go [hunk 4]
```diff
@@ -189,7 +191,9 @@ func (suite *KeeperTestSuite) TestAddPublicPlanProposal() {
 				err := keeper.HandlePublicPlanProposal(suite.ctx, suite.keeper, proposal)
 				suite.Require().NoError(err)
 
-				_, found := suite.keeper.GetPlan(suite.ctx, uint64(1))
+				plan, found := suite.keeper.GetPlan(suite.ctx, uint64(1))
+				fmt.Println(plan)
+				// TODO: need to check each field same as expected
```
Comment (kogisin) [inline x/farming/keeper/proposal_handler_test.go:196]: I'll add validation to check each field with other TODO lists in the other PR.

File: x/farming/types/plan.go [hunk 5]
```diff
@@ -307,3 +315,10 @@ func ValidateStakingCoinTotalWeights(weights sdk.DecCoins) bool {
 func IsPlanActiveAt(plan PlanI, t time.Time) bool {
 	return !plan.GetStartTime().After(t) && plan.GetEndTime().After(t)
 }
+
+func PrivatePlanFarmingPoolAddress(name string, planId uint64) sdk.AccAddress {
+	poolAddrName := strings.Join([]string{PrivatePlanFarmingPoolAddrPrefix, fmt.Sprint(planId), name}, PoolAddrSplitter)
+	// TODO: remove debugging log poolAddrName
```
[the remark this answers is not in the document]
Comment (dongsam) [reply x/farming/types/plan.go:321]: I left it for testing during the development stage, and I will get rid of todo and prints before Merge.

File: x/farming/keeper/plan.go (hunk 2 above)
Comment (dongsam) [reply x/farming/keeper/plan.go:177]: Yes, I'll add validation logic here

File: x/farming/keeper/plan.go (hunk 1 above)
Comment (dongsam) [reply x/farming/keeper/plan.go:175]: I forgot to erase todo, which was already resolved in the PR commit.
Prior to modification, there was a bug in the PublicPlan Creation where TerminationAddress was not used in AddRequestProposal and FarmingPoolAddress was used.

File: x/farming/keeper/proposal_handler_test.go (hunk 4 above)
Comment (dongsam) [reply x/farming/keeper/proposal_handler_test.go:196]: Yes Please, there was the same bug mentioned above, and which was already resolved in the PR commit.
Prior to modification, there was a bug in the PublicPlan Creation where TerminationAddress was not used in AddRequestProposal and FarmingPoolAddress was used.

File: x/farming/keeper/proposal_handler.go (hunk 3 above)
Comment (dongsam) [reply x/farming/keeper/proposal_handler.go:45]: Thank you for the suggestions, Let me modify that

**Pull request merged**